test/behavior/type.zig

```diff
@@ -808,3 +808,10 @@ test "reify enum where fields refers to part of array" {
     try testing.expect(b == .bar);
     try testing.expect(a != b);
 }
+
+test "undefined type value" {
+    const S = struct {
+        const undef_type: type = undefined;
```

Comment:
did you run into a real world example of this code? I'm thinking it would be pretty reasonable to make it a compile error if `@as(type, undefined)` is ever evaluated.

Reply:
You'd have to ask the issue author, @Rexicon226 for the original use case. I imagine there are plenty of instances of `var types: [n]type = undefined;` when reifying types to justify the value's existence, where `@splat(void)` or similar is strictly worse because you wouldn't get an error later if you try to use an uninitialized element.

Reply:
Good point about reification.